### PR TITLE
JPC: adjust UX to reflect data fetching action in Jetpack Remote Install.

### PR DESCRIPTION
**Pull request by AnnaMag**
Part of improved UX for remote install credentials submit: display Spinner when submitting site credentials and adjust submit Button label. Upon completion and before redirect to plans page, we update Button label again to reflect the status.

**To test:**
- Checkout this branch on local Calypso,
- Access `calypso.localhost:3000/jetpack/connect`
- Input site url
- Input credentials when redirected to  `calypso.localhost:3000/jetpack/connect/install`
- Verify that the Spinner and Button shows upon loading, and the latter updates label upon completion.

date fetching status:

screen shot 2018-03-14 at 20 48 56

on complete ( before redirect to the plans page):

screen shot 2018-03-14 at 14 37 44

### CODE REVIEW

File: client/jetpack-connect/remote-credentials.js [hunk 1]
```diff
@@ -102,49 +106,69 @@ export class OrgCredentialsForm extends Component {
...
 		);
 	}
 
+	renderButtonLabel() {
+		const { installError, isResponseCompleted, siteToConnect, translate } = this.props;
+		const { isSubmitting } = this.state;
+		const isFetching = ! installError && ! isResponseCompleted && isSubmitting && siteToConnect;
+
+		if ( isResponseCompleted ) {
+			return translate( 'Jetpack installed' );
```
Comment (a8ci18n) [inline client/jetpack-connect/remote-credentials.js]: Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Already have Jetpack installed?' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by seear** [comment]
Comments from testing:
* The credentials box grows in height when the spinner appears
* Probably the button should go into `disabled` state/styling while we are showing the spinner

@joanrho can help us with the positioning of the spinner.

File: client/jetpack-connect/remote-credentials.js [hunk 2]
```diff
@@ -102,49 +106,69 @@ export class OrgCredentialsForm extends Component {
...
+					{ isFetching ? <Spinner /> : null }
 				</div>
 			</div>
 		);
 	}
 
+	renderButtonLabel() {
+		const { installError, isResponseCompleted, siteToConnect, translate } = this.props;
+		const { isSubmitting } = this.state;
+		const isFetching = ! installError && ! isResponseCompleted && isSubmitting && siteToConnect;
```
Comment (seear) [inline client/jetpack-connect/remote-credentials.js]: We use this same logic twice, so you may want to make `isFetching` a prop and have this logic once inside `connect`?

File: client/jetpack-connect/remote-credentials.js [hunk 3]
```diff
@@ -102,49 +106,69 @@ export class OrgCredentialsForm extends Component {
...
 			</div>
 		);
 	}
 
+	renderButtonLabel() {
+		const { isFetching, isResponseCompleted, translate } = this.props;
+		const { isSubmitting } = this.state;
+
+		if ( isResponseCompleted ) {
+			return translate( 'Jetpack installed' );
```
Comment (a8ci18n) [inline client/jetpack-connect/remote-credentials.js]: Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Already have Jetpack installed?' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by AnnaMag** [comment]
Thanks @seear! 
Button disabled and code refactored to remove repeated logic 👍 I also tweaked the logic a bit to use `isSubmitting` to measure progress as opposed to interchange it with `isFetching` (the two mirror each other once the form is submitted).

I'll leave the styling fixes for the follow-up once we decide where to place the Spinner. 

Would appreciate another look.

File: client/jetpack-connect/remote-credentials.js [hunk 4]
```diff
@@ -102,49 +106,69 @@ export class OrgCredentialsForm extends Component {
...
 			</div>
 		);
 	}
 
+	renderButtonLabel() {
+		const { isResponseCompleted, translate } = this.props;
+		const { isSubmitting } = this.state;
+
+		if ( isResponseCompleted ) {
+			return translate( 'Jetpack installed' );
```
Comment (a8ci18n) [inline client/jetpack-connect/remote-credentials.js]: Hi! I've found a possible matching string that has already been translated **23** times:
`translate( 'Already have Jetpack installed?' )` *ES Score: 11*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by joanrho** [comment]
LGTM, but can you center the spinner in that white space between the form input field and the CTA button?

**Comment by AnnaMag** [comment]
Thanks @joanrho,  will do in a follow-up!

**Pull request merged**